apps/performance-tests/src/hierarchies/StatelessHierarchyProvider.ts

```diff
@@ -71,30 +95,35 @@ export class StatelessHierarchyProvider {
     });
   }
 
-  private createECSchemaProvider() {
-    const iModel = this._props.iModel;
+  private static createNewECSchemaProvider(iModel: IModelDb) {
     const schemas = new SchemaContext();
     const locater = new AsyncSchemaJsonLocater((schemaName) => iModel.getSchemaProps(schemaName));
     schemas.addLocater(locater);
     return createECSchemaProvider(schemas);
   }
 
   private createProvider() {
-    const schemaProvider = this.createECSchemaProvider();
-    const rowLimit = this._props.rowLimit ?? DEFAULT_ROW_LIMIT;
-    const imodelAccess = {
-      imodelKey: this._props.iModel.key,
-      ...schemaProvider,
-      ...createCachingECClassHierarchyInspector({ schemaProvider, cacheSize: 1000 }),
-      ...createLimitingECSqlQueryExecutor(createECSqlQueryExecutor(this._props.iModel), rowLimit),
-    };
+    const imodelAccess =
+      "iModel" in this._props ? StatelessHierarchyProvider.getIModelAccess(this._props.iModel, this._props.rowLimit) : this._props.imodelAccess;
     return createIModelHierarchyProvider({
       imodelAccess,
       hierarchyDefinition: this._props.getHierarchyFactory(imodelAccess),
       queryCacheSize: 0,
       filtering: this._props.filtering,
     });
   }
+
+  public static getIModelAccess(iModel: IModelDb, rowLimit?: number | "unbounded"): IModelAccess {
```

Comment:
IMO this should be named "create" rather than "get"

Reply:
Renamed